x-pack/plugins/data_enhanced/public/search/search_interceptor.ts

```diff
@@ -91,18 +122,94 @@ export class EnhancedSearchInterceptor extends SearchInterceptor {
       tap((response) => (id = response.id)),
       catchError((e: Error) => {
         cancel();
-        return throwError(this.handleSearchError(e, options, searchAbortController.isTimeout()));
+        return throwError(e);
       }),
       finalize(() => {
-        this.pendingCount$.next(this.pendingCount$.getValue() - 1);
         searchAbortController.cleanup();
-        if (untrackSearch && this.deps.session.isCurrentSession(options.sessionId)) {
-          // untrack if this search still belongs to current session
-          untrackSearch();
-        }
         if (savedToBackgroundSub) {
           savedToBackgroundSub.unsubscribe();
         }
+      }),
+      shareReplay(1)
+    );
+  }
+
+  /**
+   * @internal
+   * Creates a new search observable and a corresponding search abort controller
+   * If requestHash is defined, tries to return them first from cache.
+   */
+  private getSearchResponse$(
+    request: IKibanaSearchRequest,
+    options: IAsyncSearchOptions,
+    requestHash?: string
+  ) {
+    const cached = requestHash ? this.responseCache.get(requestHash) : undefined;
+
+    const searchAbortController =
+      cached?.searchAbortController || new SearchAbortController(this.searchTimeout);
+
+    // Create a new abort signal if one was not passed. This fake signal will never be aborted,
+    // So the underlaying search will not be aborted, even if the other consumers abort.
+    searchAbortController.addAbortSignal(options.abortSignal || new AbortController().signal);
```

Comment:
Why do we pass "fake" `new AbortController().signal` ? 
Why not `if (options.abortSignal) searchAbortController.addAbortSignal(options.abortSignal)` ?

Reply:
The comment above explains it: this makes sure that even one subscription without an abort signal keeps the controller alive.
Imagine the following scenario:
 1. Search once with a singal
 2. Search again without a signal
 3. Cancel the signal

Result: The search gets canceled
Wanted result: The search doesn't get canceled because there's one consumer that is still interested in the result.

Reply:
Got it 👍